### PR TITLE
Switch Settings.dig to dot notation

### PR DESCRIPTION
**Pull request by stevenjcumming**
## Summary

- Most settings are accessed via dot notation (`Settings.parent_key.child_key`)
- Using `.dig` can obscure the use of those keys by making them difficult to find 
- The dot notation makes them easier to find
- If dot notation isn't possible, dig is acceptable 

## Related issue(s)

- n/a

## Testing done

- [ ] n/a

## Acceptance criteria

- [ ]  Settings.dig is only used where dot notation isn't possible or difficult

### CODE REVIEW

**Comment by rmtolmach** [comment]
Wow, I just came across this today in https://github.com/department-of-veterans-affairs/vets-api/pull/18887 so I asked the PR author to change it up 🎉

**Comment by jerekshoe** [comment]
Maybe this is a silly question, but isn’t the use of `dig` in most of these to avoid errors if any of the intermediate values is nil?

**Comment by shaunburdick** [comment]
> Maybe this is a silly question, but isn’t the use of `dig` in most of these to avoid errors if any of the intermediate values is nil?

I agree with @jerekshoe on this. If we want to mandate/restrict the use of `.dig()` perhaps put out a message and/or a ticket for each team so we can validate if the `.dig` is needed or not.

**Comment by jerekshoe** [comment]
> > Maybe this is a silly question, but isn’t the use of `dig` in most of these to avoid errors if any of the intermediate values is nil?
> 
> I agree with @jerekshoe on this. If we want to mandate/restrict the use of `.dig()` perhaps put out a message and/or a ticket for each team so we can validate if the `.dig` is needed or not.

After thinking about this more I’m assuming it should be safe to use dot notation for these because they are updating usages of dig on the Settings hash, so if any of the intermediate values are nil there are probably bigger issues. I wasn’t thinking about this being scoped to the Settings hash when I asked my initial question

**Comment by stevenjcumming** [comment]
@jerekshoe @shaunburdick My opinion is the default in the context of Settings should be dot notation, and if that's not possible `dig` is fine. If a parent (intermediate) key might be nil there's two ways to go about it. 

1. safe navigation with `&`
2. `dig`

I think either are acceptable alternatives. Based on what I saw in the helm charts those values can't be nil and the settings that don't exist like `modules_appeals_api.enabled_unsafe_mode` return nil anyways

**Review by shaunburdick — DISMISSED** [review]
LGTM based on @stevenjcumming commment! 👍

File: modules/appeals_api/app/controllers/concerns/appeals_api/openid_auth.rb [hunk 1]
```diff
@@ -64,7 +64,7 @@ def token_validation_result
     private
 
     def unsafe_mode?
-      Rails.env.development? && Settings.dig(:modules_appeals_api, :enable_unsafe_mode)
+      Rails.env.development? && Settings.modules_appeals_api.enable_unsafe_mode
```
Comment (rmtolmach) [inline modules/appeals_api/app/controllers/concerns/appeals_api/openid_auth.rb:67]: @department-of-veterans-affairs/lighthouse-banana-peels This setting doesn't actually exist anywhere... It's not in settings.yml or in any of the values.yaml in the manifest repo.

File: app/controllers/v1/pension_ipf_callbacks_controller.rb [hunk 2]
```diff
@@ -63,7 +63,7 @@ def authenticity_error
     end
 
     def bearer_token_secret
-      Settings.dig(:pension_ipf_vanotify_status_callback, :bearer_token)
+      Settings.pension_ipf_vanotify_status_callback.bearer_token
```
Comment (rmtolmach) [inline app/controllers/v1/pension_ipf_callbacks_controller.rb:66]: This exists [in the manifest repo](https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/blob/main/apps/vets-api/staging/values.yaml#L2969-L2970) but not in vets-api settings.yml. 🤔 They were added to `config/settings/test.yml`, but I think it was an oversight that they weren't added to settings.yml. Can you do that as part of this PR? Or ask the original PR author to? https://github.com/department-of-veterans-affairs/vets-api/issues/15883

File: app/controllers/v1/nod_callbacks_controller.rb [hunk 3]
```diff
@@ -80,7 +80,7 @@ def authenticity_error
     end
 
     def bearer_token_secret
-      Settings.dig(:nod_vanotify_status_callback, :bearer_token)
+      Settings.nod_vanotify_status_callback.bearer_token
```
Comment (rmtolmach) [inline app/controllers/v1/nod_callbacks_controller.rb:83]: This setting exists in the manifest repo and `config/settings/test.yml`, but not settings.yml (i.e. same issue as `Settings.pension_ipf_vanotify_status_callback.bearer_token`).

**Review by rmtolmach — COMMENTED** [review]
Overall I like this! I found a few issues with some of the settings though.

**Comment by stevenjcumming** [comment]
@rmtolmach I've add 

`pension_ipf_vanotify_status_callback` and `nod_vanotify_status_callback` bearer_tokens to `settings.yml`

**Review by rmtolmach — APPROVED** [review]

**Pull request merged**